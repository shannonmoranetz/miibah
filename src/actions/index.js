export const SET_AMIIBOS = 'SET_AMIIBOS';
export const ADD_TO_WISHLIST = 'ADD_TO_WISHLIST';
export const ADD_TO_COLLECTED = 'ADD_TO_COLLECTED';
export const GET_WISHLIST = 'GET_WISHLIST';
export const GET_COLLECTED = 'GET_COLLECTED';
export const SET_LOADING = 'SET_LOADING';
export const SET_ERROR = 'SET_ERROR';

export const setAmiibos = amiibos => ({
  type: SET_AMIIBOS,
  amiibos
});

export const addToWishlist = amiibo => ({
  type: ADD_TO_WISHLIST,
  amiibo
});

export const addToCollected = amiibo => ({
  type: ADD_TO_COLLECTED,
  amiibo
});

export const getWishlist = wishlist => ({
  type: GET_WISHLIST,
  wishlist
});

export const getCollected = collected => ({
  type: GET_COLLECTED,
  collected
});

export const setLoading = isLoading => ({
  type: SET_LOADING,
  isLoading
});

export const setError = error => ({
  type: SET_ERROR,
  error
});

