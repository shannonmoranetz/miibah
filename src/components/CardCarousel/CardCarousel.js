import React, { Component } from 'react';
import { connect } from 'react-redux';
import Card from '../../containers/Card/Card.js';

export class CardCarousel extends Component {

  returnAmiibos = (amiibos) => {
      return amiibos.map((amiibo, i) => {
      return <Card name = {amiibo.name}
                   amiiboSeries = {amiibo.amiiboSeries}
                   gameSeries = {amiibo.gameSeries}
                   image = {amiibo.image}
                   release = {amiibo.release.na}
                   id = {amiibo.id}
                   key = {i} />
    })
  }

  render() {
    console.log(this.props)
    return (
      <div className="CardCarousel">
        { this.props.match.path === '/' && this.returnAmiibos(this.props.amiibos) }
        { this.props.match.path === '/amiibo/:id' && this.returnAmiibos(this.props.amiibos) }
      </div>
    )
  }
}

export const mapStateToProps = state => ({
  amiibos: state.amiibos
});

export default connect(mapStateToProps)(CardCarousel);
