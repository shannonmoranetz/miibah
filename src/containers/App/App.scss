html {
  height: 100%;
  width: 100%;
}

body {
  background-image: url("images/background.jpg");
  background-attachment: fixed;
  background-repeat: no-repeat;
  background-size: cover;
}

.load-text {
  color: white;
  text-align: center;
  font-size: 50px;
  margin: 60px auto;
  font-family: $heading-font;
}