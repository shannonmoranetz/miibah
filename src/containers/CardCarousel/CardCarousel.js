import React, { Component } from 'react';
import { connect } from 'react-redux';
import Card from '../../components/Card/Card.js';

export class CardCarousel extends Component {

  returnAmiibos = (amiibos) => {
      return amiibos.map((amiibo, i) => {
        return <Card  name = {amiibo.name}
                      amiiboSeries = {amiibo.amiiboSeries}
                      gameSeries = {amiibo.gameSeries}
                      image = {amiibo.image}
                      release = {amiibo.release.na}
                      key = {i} />
    })
  }

  render() {
    return (
      <div className="CardCarousel">
        { this.props.match.path === '/' && this.returnAmiibos(this.props.amiibos) }
        { this.props.match.path === '/wishlist' && this.returnAmiibos(this.props.wishlist) }
        { this.props.match.path === '/collected' && this.returnAmiibos(this.props.collected) }
        { this.props.match.path === '/amiibo/:name' && this.returnAmiibos(this.props.amiibos) }
        { this.props.match.path === '/search/:name' && this.returnAmiibos(this.props.searchedAmiibo) }
      </div>
    )
  }
}

export const mapStateToProps = state => ({
  amiibos: state.amiibos,
  searchedAmiibo: state.searchedAmiibo,
  wishlist: state.wishlist,
  collected: state.collected
});

export default connect(mapStateToProps)(CardCarousel);
