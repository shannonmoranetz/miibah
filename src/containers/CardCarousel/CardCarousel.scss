.CardCarousel {
  display: flex;
  overflow: scroll;
  overflow-y: hidden;
  background-color: $brightpink;
  margin: 20px 0px;
  border-top: 20px solid $darkblue;
  border-bottom: 20px solid $darkblue;
}

.empty-message {
  font-family: $heading-font;
  color: white;
  text-align: center;
  font-size: 30px;
}