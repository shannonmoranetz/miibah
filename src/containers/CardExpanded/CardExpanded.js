import React, { Component } from 'react';

class CardExpanded extends Component {
  render() {
    return (
      <div className="CardExpanded">
        <div className="innerCard">
          <div className="cardImageContainer">
            <img className="previewImageExpanded" src={this.props.image} alt={`${this.props.name} amiibo`}/>
          </div>
          <div className="cardInfo">
            <h2 className="cardName">{this.props.name}</h2>
            <p className="cardAmiiboSeries"> amiibo series: {this.props.amiiboSeries}</p>
            <p className="cardGameSeries">game series: {this.props.gameSeries}</p>
            <p className="cardDate">release date: {this.props.release.na}</p>
          </div>
      </div>
    </div>
    );
  }
}

export default CardExpanded;
