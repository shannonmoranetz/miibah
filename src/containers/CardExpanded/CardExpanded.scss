.CardExpanded {
  z-index: 1;
  position: absolute;
  top: 25%;
  right: 0;
  bottom: 0;
  left: 0;
}

.inner-card {
  margin: 0px auto;
  border: 6px solid $darkblue;
  display: inline-block;
  font-family: $body-font;
  width: 80%;
  color: white;
  background-color: $ice;
  display: flex;
  justify-content: space-around;
  align-items: center;
  text-transform: lowercase;
}

.card-info {
  margin-top: auto;
  margin-bottom: auto;
}

.card-name {
  font-family: $heading-font;
  font-size: 40px;
}

.image-expanded {
  margin: 30px auto;
  height: 300px;
}

.card-amiibo-series,
.card-game-series,
.card-date {
  font-size: 25px;
}

.exit-button {
  margin-bottom: 30%;
  text-align: center;
  padding: 5px 15px;
  font-size: 30px;
  border: 2px solid white;
  background-color: $salmon;
}