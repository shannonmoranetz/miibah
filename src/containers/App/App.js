import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getAmiibos } from '../../thunks/getAmiibos.js';
import { withRouter, Route, Switch } from 'react-router-dom';
import { getWishlist, getCollected }  from '../../actions/';
import CardCarousel from '../../containers/CardCarousel/CardCarousel.js';
import Header from '../../components/Header/Header.js';
import Menu from '../../components/Menu/Menu.js';
import ErrorMessage from '../../components/ErrorMessage/ErrorMessage.js';
import NoResultsMessage from '../../components/NoResultsMessage/NoResultsMessage.js';
import SearchBar from '../../containers/SearchBar/SearchBar.js';
import CardExpanded from '../../containers/CardExpanded/CardExpanded.js';
import PropTypes from 'prop-types';

export class App extends Component {
  componentDidMount = () => {
    const url = 'http://www.amiiboapi.com/api/amiibo/';
    this.props.getAmiibos(url);
    this.populateWishlist();
    this.populateCollected();
    this.props.history.push('/');
  }

  populateWishlist = () => {
    const wishlist = localStorage.getItem('wishlist');
    const parsedWishlist = JSON.parse(wishlist);
    if (localStorage.getItem('wishlist') !== null) {
      this.props.getWishlist(parsedWishlist);
    }
  }

  populateCollected = () => {
    const collected = localStorage.getItem('collected');
    const parsedCollected = JSON.parse(collected);
    if (localStorage.getItem('collected') !== null) {
      this.props.getCollected(parsedCollected);
    }
  }
  
  findAmiibo = ({ match, history }) => {
    const { amiibos } = this.props;
    const amiibo = amiibos.find(amiibo => amiibo.name === match.params.name);
    return amiibo ? ([
      <CardCarousel match={match} history={history} />,
      <CardExpanded amiibo={amiibo} match={match} history={history} />
    ]) : <ErrorMessage />
  }

  render() {
    return (
      <div className="App">
        <Header />
        { this.props.isLoading && <h2 className="load-text">loading...</h2> }
        { !this.props.error ? (
            <div className="inner-app">
              <Menu />
              <SearchBar />
                <Switch>
                  <Route path='/amiibo/:name' render={this.findAmiibo} />
                  <Route path='/wishlist/:name' render={this.findAmiibo} />
                  <Route path='/collected/:name' render={this.findAmiibo} />
                  <Route path='/search/:name' render={() => <CardExpanded amiibo={this.props.searchedAmiibo[0]} history={this.props.history}/>} />
                  <Route path='/wishlist' component={CardCarousel} />
                  <Route path='/collected' component={CardCarousel} />
                  <Route exact path='/' component={CardCarousel} />
                  <Route component={ErrorMessage} />
                </Switch>
            </div>
        ) : (
          <Route component={NoResultsMessage} />
        )}
      </div>
    )
  }
}

export const mapStateToProps = state => ({
  amiibos: state.amiibos,
  searchedAmiibo: state.searchedAmiibo,
  isLoading: state.isLoading,
  error: state.error
});

export const mapDispatchToProps = dispatch => ({
  getAmiibos: (url) => dispatch(getAmiibos(url)),
  getWishlist: (wishlist) => dispatch(getWishlist(wishlist)),
  getCollected: (collected) => dispatch(getCollected(collected))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));

App.propTypes = {
  isLoading: PropTypes.bool,
  getAmiibos: PropTypes.func,
  getWishlist: PropTypes.func,
  getCollected: PropTypes.func,
  history: PropTypes.object,
  location: PropTypes.object,
  match: PropTypes.object,
  amiibos: PropTypes.array
}