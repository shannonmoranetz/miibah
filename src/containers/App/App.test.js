import React from 'react';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { getAmiibos } from '../../thunks/getAmiibos.js';
import { getWishlist, getCollected } from '../../actions';
import { shallow } from 'enzyme';
import CardExpanded from '../CardExpanded/CardExpanded.js';
import CardCarousel from '../../components/CardCarousel/CardCarousel.js';

jest.mock('../../thunks/getAmiibos.js');

const matchMock = { params: { id: 1 } };

const propsMock = {
  getAmiibos: jest.fn(),
  getWishlist: jest.fn(),
  getCollected: jest.fn(),
  amiibos: [{ name: 'pikachu', id: 1 }]
}

describe('App', () => {
  let wrapper; 
  beforeEach(() => {
    wrapper = shallow(<App {...propsMock} />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call getAmiibos when App is mounted', () => {
    wrapper.instance().componentDidMount();
    expect(propsMock.getAmiibos).toHaveBeenCalled();
  });

  it('should call getWishlist with correct params when populateWishList is called', () => {
    localStorage.setItem('wishlist', JSON.stringify([{name: 'pikachu'}]));
    wrapper.instance().populateWishlist();
    expect(propsMock.getWishlist).toHaveBeenCalledWith([{name: 'pikachu'}]);
  });

  it('should call getCollected with correct params when populateWishList is called', () => {
    localStorage.setItem('collected', JSON.stringify([{name: 'squirtle'}]));
    wrapper.instance().populateCollected();
    expect(propsMock.getCollected).toHaveBeenCalledWith([{name: 'squirtle'}]);
  });

  it('should return the matched amiibo', () => {
    const amiibo = { id: 1, name: 'pikachu' };
    const expected = [<CardCarousel />, <CardExpanded {...amiibo} match={matchMock} />]
    const result = wrapper.instance().findAmiibo({ match: { params: { id: 1 } } });
    expect(result).toEqual(expected);
  });









  describe('mapStateToProps', () => {
    it('should return a props object with an amiibos array and loading boolean', () => {
      const expected = { amiibos: [{id: 1}, {id: 2}], isLoading: false };
      const mockState = { amiibos: [{id: 1}, {id: 2}], isLoading: false, favorites: [] };
      const props = mapStateToProps(mockState);
      expect(props).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    const dispatchMock = jest.fn();
    const props = mapDispatchToProps(dispatchMock);
    it('should call dispatch with getAmiibos as a param', () => {
        const expected = getAmiibos();
        props.getAmiibos();
        expect(dispatchMock).toHaveBeenCalledWith(expected);
    });

    it('should call dispatch with getWishlist as a param', () => {
      const expected = getWishlist([{ id: 1 }, { id: 2 }]);
      props.getWishlist([{ id: 1 }, { id: 2 }]);
      expect(dispatchMock).toHaveBeenCalledWith(expected);
    });

    it('should call dispatch with getCollected as a param', () => {
      const expected = getCollected([{ id: 1 }, { id: 2 }]);
      props.getCollected([{ id: 1 }, { id: 2 }]);
      expect(dispatchMock).toHaveBeenCalledWith(expected);
    });
  });

});

