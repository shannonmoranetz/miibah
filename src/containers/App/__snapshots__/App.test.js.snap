// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <Header />
  <div
    className="inner-app"
  >
    <Menu />
    <withRouter(Connect(SearchBar)) />
    <Switch>
      <Route
        path="/amiibo/:name"
        render={[Function]}
      />
      <Route
        path="/wishlist/:name"
        render={[Function]}
      />
      <Route
        path="/collected/:name"
        render={[Function]}
      />
      <Route
        path="/search/:name"
        render={[Function]}
      />
      <Route
        component={[Function]}
        path="/wishlist"
      />
      <Route
        component={[Function]}
        path="/collected"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </div>
</div>
`;
